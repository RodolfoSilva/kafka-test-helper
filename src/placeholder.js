export const WHAT_IS_THAT_MESSAGE_KEY = 'what_is_that_message'
export const WHAT_IS_THAT_MESSAGE_VALUE = 'https://github.com/chrvadala/kafka-test-helper'

export function makePlaceholderMessages(uuid, partions) {
    const messages = []
    for(let i =0; i<partions; i++){
        messages.push({
            partition: i,
            value: JSON.stringify({
                [WHAT_IS_THAT_MESSAGE_KEY]: WHAT_IS_THAT_MESSAGE_VALUE,
                uuid
            })
        })
    }

    return messages;
}

export function isPlaceholderMessageWithUUID(message, uuid) {
    let json;

    try{
        json=JSON.parse(message)
    }catch{
        return false
    }

    return typeof json === 'object' 
        && json !== null 
        && json.hasOwnProperty('uuid')
         && json.uuid === uuid
}

