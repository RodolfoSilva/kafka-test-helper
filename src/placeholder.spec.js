import { makePlaceholderMessages, isPlaceholderMessageWithUUID } from './placeholder.js'

describe.only("placeholder", () => {

    test("makePlaceholderMessages", () => {
        const placeholder = JSON.stringify({
            what_is_that_message: 'https://github.com/chrvadala/kafka-test-helper',
            uuid: 'test123'
        })

        const messages = makePlaceholderMessages('test123', 4);

        const expected = [
            { partition: 0, value: placeholder },
            { partition: 1, value: placeholder },
            { partition: 2, value: placeholder },
            { partition: 3, value: placeholder },
        ]

        expect(messages).toMatchSnapshot()
        expect(messages).toEqual(expected)
    })


    test("isPlaceholderMessageWithUUID", () => {
        const placeholder = JSON.stringify({
            what_is_that_message: 'https://github.com/chrvadala/kafka-test-helper',
            uuid: 'test123'
        })

        const invalidPlaceholder = JSON.stringify({
            bar: "abc",
            foo: 42
        })


        expect(isPlaceholderMessageWithUUID(placeholder, 'test123')).toBe(true)
        expect(isPlaceholderMessageWithUUID(placeholder, 'testabc')).toBe(false)

        expect(isPlaceholderMessageWithUUID(invalidPlaceholder, 'test123')).toBe(false)

        expect(isPlaceholderMessageWithUUID("tfsahkh{", 'test123')).toBe(false)
        expect(isPlaceholderMessageWithUUID(null, 'test123')).toBe(false)
    })
})