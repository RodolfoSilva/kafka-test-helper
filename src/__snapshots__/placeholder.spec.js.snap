// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`placeholder makePlaceholderMessages 1`] = `
Array [
  Object {
    "partition": 0,
    "value": "{\\"what_is_that_message\\":\\"https://github.com/chrvadala/kafka-test-helper\\",\\"uuid\\":\\"test123\\"}",
  },
  Object {
    "partition": 1,
    "value": "{\\"what_is_that_message\\":\\"https://github.com/chrvadala/kafka-test-helper\\",\\"uuid\\":\\"test123\\"}",
  },
  Object {
    "partition": 2,
    "value": "{\\"what_is_that_message\\":\\"https://github.com/chrvadala/kafka-test-helper\\",\\"uuid\\":\\"test123\\"}",
  },
  Object {
    "partition": 3,
    "value": "{\\"what_is_that_message\\":\\"https://github.com/chrvadala/kafka-test-helper\\",\\"uuid\\":\\"test123\\"}",
  },
]
`;
